src/Core/Services/ResolverMiddleware.cs

```diff
@@ -265,6 +265,8 @@ protected static bool IsInnerObject(IMiddlewareContext context)
                 SINGLE_TYPE => ((FloatValueNode)value).ToSingle(),
                 FLOAT_TYPE => ((FloatValueNode)value).ToDouble(),
                 DECIMAL_TYPE => ((FloatValueNode)value).ToDecimal(),
+                // If we reach here, we can be sure that the value will not be null.
+                ID_TYPE => Guid.Parse(value.Value!.ToString()!),
```

Comment:
Do we have tests for CosmosDB that check the Guid type? If yes, are those affected in any way. Just checking to ensure there are no regressions.

Reply:
Cosmos DB has an entity called Planet, which has ID of Guid type.

Reply:
Continuing conversation here: https://github.com/Azure/data-api-builder/pull/1644#discussion_r1304146910